code/modules/paperwork/ticketmachine.dm

```diff
@@ -31,6 +31,11 @@
 /obj/machinery/ticket_machine/Initialize(mapload)
 	. = ..()
 	update_appearance()
+	///Standard wall-hung object behavior.
+	var/turf/attachable_wall = get_step(src, dir)
+	if(!iswallturf(attachable_wall))
+		return //Nothing to latch onto.
+	attachable_wall.AddComponent(/datum/component/wall_link, src)
```

Comment:
this pattern is copy pasted quite often, could it be rolled into a proc that's overridden or someodd

Reply:
I moved it to a proc on obj for the meantime, I think this should be okay for now but to be on the safe side you tell me if it's cool to have that kind of high-level component assignment as opposed to using it as a getter for the turf and then assigning it on object init